### ISSUE TITLE
taos_query never returns

### ISSUE THREAD

**Issue by effun** [taosdata/TDengine#603]
**Bug Description**
I'm using taos.dll for my .net application to insert massive data into TDenging database. Everything looks fine as the application started. But after running for about 46 hours, the application hangs up. By stepping into the code, I figured out the problem was caused by calling taos_query method in taos.dll, taos_query blocked like forever.
I've tried to solve this problem by restarting the application, it doesn't work, taos_query blocked at the first call. Restarting TDenging database can solve it, but with serious side effect witch is another issue I'll discuss later.

英文不好，我用中文再重述一遍。
我在我的.net应用程序里封装了taos.dll作为客户端连接器访问TDenging数据库，主要用于大量数据的写入。在刚开始运行的时候一切看似正常，但当运行了大约46小时以后，程序卡壳了。通过调试我发现，是由于调用taos.dll中的taos_query方法，这个方法长时间（永远）不返回，造成整个应用卡壳，后续数据无法再写入。
我曾试着重启应用程序来解决问题，但没用，程序在第一次调用taos_query方法的时候就卡住了。只有重启TDengine数据库才能解决这个问题，但产生了严重的后果-大量数据丢失，这是另外一个问题，我将晚些再来讨论。

**To Reproduce**
1. Insert massive data into database continuously by calling taos_query in one connection. (800 records per second)
2. Wait for at least 46 hours.
3. Application hangs.

**Environment (please complete the following information):**
 - OS: Windows Server 2012 Standard 64bit
 - Memory 32G, CPU Intel Xeon E5-2630
 - TDengine Version [e.g. 1.6.6.2]

**Comment by effun** [comment]
by digging around, I figured out that the problem may not be caused by client component, but the server it self.
I've wrote a simple test application and ran it on the same machine. It can run `SELECT` statement successfully, but blocked on 'INSERT' statement.
During the process, an error occurred unexpectedly, which told me the database file corrupted.
`taos> select * from v_2_00H10038 ;
Query interrupted (DB file corrupted), 0 row(s) in set (0.002000s)`
So, I think it is something related to the issue I mentioned above.

**Comment by hjxilinx** [comment]
It seems that some of the  database files are broken. Are the data  "IMPORT" or "INSERT" into TDengine?

**Comment by effun** [comment]
Data is INSERTed into database. Each statement contains at most 500 rows for 500 sub tables of one single super table. Execution rate is 700 - 2000 rows per second. So there may be 2 or 3 insert statements executed in one second.
The insert statement looks like below:
`INSERT INTO sub_table1 USING super_table TAGS('tag1') VALUES (...)`
`     sub_table2 USING super_table TAGS('tag2') VALUES (...)`
`     ... ;'

**Comment by effun** [comment]
There's one more thing maybe important. I don't guarantee all data inserted in one thread, but all threads shares the same connection returned by `taos_connect`.

**Comment by plum-lihui** [comment]
能加微信或qq详细交流一下吗？最好能让我们远程定位一下。多谢！

**Comment by effun** [comment]
实在抱歉，现在已经改用其它数据库，代码也全部改了，要重现问题比较不现实。而且我不确定是不是我的.net封装造成的问题，如果你们需要继续研究的，我可以把封装的源代码发给你。

**Comment by plum-lihui** [comment]
好的，请将测试的源码发送给我们吧：huili@taosdata.com。多谢！

**Comment by effun** [comment]
我会写个模拟程序来尝试重现问题，需要花点时间，封装的代码我稍后就发给你。

**Comment by sangshuduo** [comment]
@effun  请问还有问题吗？

**Comment by effun** [comment]
没有了，谢谢
